store/sql_channel_store.go

```diff
@@ -659,6 +659,8 @@ func (s SqlChannelStore) UpdateMember(member *model.ChannelMember) StoreChannel
 			result.Data = member
 		}
 
+		s.InvalidateAllChannelMembersForUser(member.UserId)
```

Comment:
This can't be done here. You need to update the HA cluster as well. See the other caches for examples.